pkg/cgroups/cgroups_linux.go

```diff
@@ -45,6 +52,49 @@ func mountCgroup() error {
 	return nil
 }
 
+// cgrpSanityCheck performs a sanity check on the mounted cgroup fs and issues
+// a warning if something seems wrong.
+//
+// It does this by recursively checking the number of pids in "cgroup.procs". If they are 2
+// or less, it is likely that the cilium pod is the only container in this
+// cgroup.
+func cgrpSanityCheck() {
+
+	procCount := 0
+	err := filepath.Walk(cgroupRoot, func(path string, info fs.FileInfo, _ error) error {
```

Comment:
>It does this by recursively checking the number of pids in "cgroup.procs".

I don't think I quite follow the motivation, and the checks this PR is introducing. For example, how does this work in Kind like cgroup hierarchies? 

>}).Warn("Process count under cgroup is unexpectedly low")

Also, how do users interpret this warning? What actions are they supposed to take? What are the implications?

Reply:
I see that the documentation gives some more details. But I wonder if the warning messages themselves should provide these details instead?

Reply:
> > It does this by recursively checking the number of pids in "cgroup.procs".
> 
> I don't think I quite follow the motivation, and the checks this PR is introducing. For example, how does this work in Kind like cgroup hierarchies?

What are kind-like cgroup hierarchies? I don't see how running this on kind would change things. The idea here is to catch the case where cilium tries to use a cgroup that only includes the cilium agent, and no other process.

> 
> > }).Warn("Process count under cgroup is unexpectedly low")
> 
> Also, how do users interpret this warning? What actions are they supposed to take? What are the implications?

The warning points to the docs, where hopefully describes all the above. Not clear to me if you think that the documentation is not clear enough or it was not clear that the message point to the docs.